flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java

```diff
@@ -236,6 +252,213 @@ void testCreateMaterializedTableInFullMode() {
                         "Only support create materialized table in continuous refresh mode currently.");
     }
 
+    @Test
+    void testAlterMaterializedTableRefresh() throws Exception {
+        long timeout = Duration.ofSeconds(20).toMillis();
+        long pause = Duration.ofSeconds(2).toMillis();
+        // initialize session handle, create test-filesystem catalog and register it to catalog
+        // store
+        SessionHandle sessionHandle = initializeSession();
+
+        List<Row> data = new ArrayList<>();
+        data.add(Row.of(1L, 1L, 1L, "2024-01-01"));
+        data.add(Row.of(2L, 2L, 2L, "2024-01-02"));
+        data.add(Row.of(3L, 3L, 3L, "2024-01-02"));
+        String dataId = TestValuesTableFactory.registerData(data);
+
+        String sourceDdl =
+                String.format(
+                        "CREATE TABLE my_source (\n"
+                                + "  order_id BIGINT,\n"
+                                + "  user_id BIGINT,\n"
+                                + "  shop_id BIGINT,\n"
+                                + "  order_created_at STRING\n"
+                                + ")\n"
+                                + "WITH (\n"
+                                + "  'connector' = 'values',\n"
+                                + "  'bounded' = 'true',\n"
+                                + "  'data-id' = '%s'\n"
+                                + ")",
+                        dataId);
+        service.executeStatement(sessionHandle, sourceDdl, -1, new Configuration());
```

Comment:
we should wait until this statement is finished, then we can create the materialized table.